src/typing/buildMacro.ml

```diff
@@ -0,0 +1,148 @@
+open Ast
+open Common
+open Type
+open Typecore
+open Error
+
+
+type candidate =
+	| Candidate of pos * path * string * recover_mode
+
+and recover_mode =
+	| RMBoth
+	| RMMethod
+	| RMModule
+	| RMNo
+
+
+let candidate_pos  = function
+	| Candidate(p,_,_,_) -> p
+
+let apply_macro ctx mode path el p =
+	let rev_path = List.rev path in
+	let starts_with_uppercase s =
+		if String.length s > 0 then
+			let first_char = String.get s 0 in
+			(Char.lowercase first_char) != first_char
+		else
+			false
+	in
+	let check_candidates candidates try_absolute_path  =
+		let sort_candidates c =
```

Comment:
I get that you would sort by position to determine the order within a given file. But this seems to also consider the position pfile. Are you sure this is the right thing to do? Import order is significant and should not be ignored.